gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GCSManualClientImpl.java

```diff
@@ -0,0 +1,222 @@
+    return gcsClientDelegate.listBucketNames();
+  }
+
+  @Override
+  public List<GoogleCloudStorageItemInfo> listBucketInfo() throws IOException {
+    return gcsClientDelegate.listBucketInfo();
+  }
+
+  @Override
+  public List<GoogleCloudStorageItemInfo> listObjectInfo(
+      String bucketName, String objectNamePrefix, ListObjectOptions listOptions)
+      throws IOException {
+    return gcsClientDelegate.listObjectInfo(bucketName, objectNamePrefix, listOptions);
+  }
+
+  @Override
+  public ListPage<GoogleCloudStorageItemInfo> listObjectInfoPage(
+      String bucketName, String objectNamePrefix, ListObjectOptions listOptions, String pageToken)
+      throws IOException {
+    return gcsClientDelegate.listObjectInfoPage(
+        bucketName, objectNamePrefix, listOptions, pageToken);
+  }
+
+  @Override
+  public GoogleCloudStorageItemInfo getItemInfo(StorageResourceId resourceId) throws IOException {
+    return gcsClientDelegate.getItemInfo(resourceId);
+  }
+
+  @Override
+  public List<GoogleCloudStorageItemInfo> getItemInfos(List<StorageResourceId> resourceIds)
+      throws IOException {
+    return gcsClientDelegate.getItemInfos(resourceIds);
+  }
+
+  @Override
+  public List<GoogleCloudStorageItemInfo> updateItems(List<UpdatableItemInfo> itemInfoList)
+      throws IOException {
+    return gcsClientDelegate.updateItems(itemInfoList);
+  }
+
+  @Override
+  public void compose(
+      String bucketName, List<String> sources, String destination, String contentType)
+      throws IOException {
+    gcsClientDelegate.compose(bucketName, sources, destination, contentType);
+  }
+
+  @Override
+  public GoogleCloudStorageItemInfo composeObjects(
+      List<StorageResourceId> sources, StorageResourceId destination, CreateObjectOptions options)
+      throws IOException {
+    return gcsClientDelegate.composeObjects(sources, destination, options);
+  }
+
+  @Override
+  public Map<String, Long> getStatistics() {
+    return gcsClientDelegate.getStatistics();
+  }
+
+  @Override
+  public void close() {
+    gcsClientDelegate.close();
+  }
+
+  public static class GCSManualClientImplBuilder {
+
+    private Credentials credentials;
+    private com.google.api.services.storage.Storage storage;
+    private HttpRequestInitializer httpRequestInitializer;
+    private GoogleCloudStorageOptions storageOptions;
+    private Function<List<AccessBoundary>, String> downscopedAccessTokenFn;
+
+    public GCSManualClientImplBuilder(
+        GoogleCloudStorageOptions storageOptions,
+        Credentials credentials,
+        Function<List<AccessBoundary>, String> downscopedAccessTokenFn) {
+      this.storageOptions = storageOptions;
+      this.credentials = credentials;
+      this.downscopedAccessTokenFn = downscopedAccessTokenFn;
+    }
+
+    @VisibleForTesting
+    public GCSManualClientImplBuilder withApairyClientStorage(
+        com.google.api.services.storage.Storage storage) {
+      this.storage = storage;
```

Comment:
can validate this to be non-null. For others as well

Reply:
yeah, there is no harm doing it here.